google-cloud-firestore/lib/google/cloud/firestore/commit_response.rb

```diff
@@ -0,0 +1,139 @@
+      #
+      #   commit_response = firestore.batch do |b|
+      #     # Set the data for NYC
+      #     b.set("cities/NYC", { name: "New York City" })
+      #
+      #     # Update the population for SF
+      #     b.update("cities/SF", { population: 1000000 })
+      #
+      #     # Delete LA
+      #     b.delete("cities/LA")
+      #   end
+      #
+      #   puts commit_response.commit_time
+      #   commit_response.write_results.each do |write_result|
+      #     puts write_result.update_time
+      #     puts write_result.transform_results.inspect
+      #   end
+      #
+      class CommitResponse
+        ##
+        # The time at which the commit occurred.
+        #
+        # @return [Time] The commit time.
+        attr_accessor :commit_time
+
+        ##
+        # The result of applying the writes.
+        #
+        # This i-th write result corresponds to the i-th write in the request.
+        #
+        # @return [Array<CommitResponse::WriteResult>] The write results.
+        attr_accessor :write_results
+
+        ##
+        # @private
+        def self.from_grpc grpc, client
+          commit_time = Convert.timestamp_to_time grpc.commit_time
+          write_results = Array(grpc.write_results).map do |write_result|
+            WriteResult.from_grpc write_result, client
+          end
+
+          new.tap do |resp|
+            resp.instance_variable_set :@commit_time,   commit_time
+            resp.instance_variable_set :@write_results, write_results
+          end
+        end
+
+        ##
+        # # WriteResult
+        #
+        # Represents the result of applying a write.
+        #
+        # @example
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   commit_response = firestore.batch do |b|
+        #     # Set the data for NYC
+        #     b.set("cities/NYC", { name: "New York City" })
+        #
+        #     # Update the population for SF
+        #     b.update("cities/SF", { population: 1000000 })
+        #
+        #     # Delete LA
+        #     b.delete("cities/LA")
+        #   end
+        #
+        #   puts commit_response.commit_time
+        #   commit_response.write_results.each do |write_result|
+        #     puts write_result.update_time
+        #     puts write_result.transform_results.inspect
+        #   end
+        #
+        class WriteResult
+          ##
+          # The last update time of the document after applying the write. Not
+          # set after a +delete+.
+          #
+          # If the write did not actually change the document, this will be
+          # the previous update_time.
+          #
+          # @return [Time] The last update time.
+          attr_accessor :update_time
+
+          ##
+          # The results of applying each change, in the order given.
+          #
+          # @return [Array<Object>] The change results.
+          attr_accessor :transform_results
```

Comment:
The way we have structured the API, our users are not actually exposed to the difference between a normal write and a document transform. It feels slightly weird to break this here and expose transformation results.

Reply:
So the only think WriteResult has is `update_time`? Would it be possible to just return the time then? And it looks like the [node implementation](https://github.com/googleapis/nodejs-firestore/blob/master/src/write-batch.js#L549) sets `update_time` to `commit_time` when `update_time` is missing. Is that right?